lmcache/v1/storage_backend/connector/valkey_connector.py

```
@@ -0,0 +1,382 @@
+            self.loop,
+        )
+        return future.result()
+
+    async def _get(self, key: CacheEngineKey) -> Optional[MemoryObj]:
+        metadata_key, kv_key = self._get_keys(key)
+
+        results = await self.connection.mget([metadata_key, kv_key])
+
+        if len(results) != 2:
+            return None
+
+        metadata_bytes, kv_bytes = results[0], results[1]
+
+        if metadata_bytes is None:
+            return None
+
+        assert not inspect.isawaitable(metadata_bytes)
+
+        metadata = RemoteMetadata.deserialize(memoryview(metadata_bytes))
+
+        memory_obj = self.local_cpu_backend.allocate(
+            metadata.shape,
+            metadata.dtype,
+            metadata.fmt,
+        )
+        if memory_obj is None:
+            logger.warning("Failed to allocate memory during remote receive")
+            return None
+
+        if kv_bytes is None:
+            logger.warning(
+                "Key exists but KV cache does not exist."
+                "Might happen when the cache is evicted by valkey."
+            )
+            memory_obj.ref_count_down()
+            return None
+
+        assert not inspect.isawaitable(kv_bytes)
+
+        try:
+            if isinstance(memory_obj.byte_array, memoryview):
+                view = memory_obj.byte_array
+                if view.format == "<B":
+                    view = view.cast("B")
+            else:
+                view = memoryview(memory_obj.byte_array)
+
+            if isinstance(kv_bytes, (bytes, bytearray)):
+                view[: metadata.length] = kv_bytes
+            elif isinstance(kv_bytes, str):
+                converted = kv_bytes.encode("utf-8")
+                view[: metadata.length] = converted
+            else:
+                converted = bytes(kv_bytes)
+                view[: metadata.length] = converted
+        except Exception as exc:
+            logger.error(f"Fail to converting : {exc}")
```

@bluayer The way you swallow the exception, it mean that the caller will not aware this exception occurred, suggest to return None.

---

I added `return None` at recent commit.